docs/weave-tutorial/02-domain.md

```diff
@@ -0,0 +1,30 @@
+---
+id: domain 
+title: Defining the Domain 
+sidebar_label: Domain 
+---
+
+The first thing we consider is the data we want to store (the state). After that, we can focus on the messages, which trigger state transitions. All blockchain state must be stored in our merkle-ized store, that provide validity hashes and proofs. This is exposed to the application as a basic key-value store, which also allows in-order iteration over the keys. On top of this, we have built some tools like secondary indexes and sequences, similar to how [storm adds an ORM](https://github.com/asdine/storm#simple-crud-system) on top of [boltdb’s key-value store](https://github.com/boltdb/bolt#using-buckets). We have avoided struct tags and tried to type as strictly as we can (without using generics).
+
+We are building an Orderbook application. This topic is quite hot at the time of writing this tutorial.
+
+- **Markets** will contain rules as to who (which public keys) may create an **order book** or who can cancel an order. The blockchain may have multiple markets. Each market may have multiple orderbooks. Each token pair can only have one orderbook per market.
+There is no global chain owner, but each market has one that adds order books and sets fees. This could be one person, a multisig, or a governance contract (DAO). When adding an order  book, the market with a given market ID is checked, then look up the owner *for that market*. Rather than having one owner that can create order books for all markets, each market stores who can update it.
```

Comment:
```suggestion
There is no global chain owner, but each market has its own chain owner that adds order books and sets fees. This could be one person, a multisig, or a governance contract (DAO). When adding an order book, the market with a given market ID is checked, then looks up the owner *for that market*. Rather than having one owner that can create order books for all markets, each market stores the information on which owner can update it.
```